### ISSUE TITLE
Touchscreen mouse speed much slower vertically than horizontally?

### ISSUE THREAD

**Issue by loomes77** [libretro/libretro-uae#614]
Hi guys, I've got a little problem with the mouse speed: I have an RP2+ and want to control the mouse pointer with its touchscreen. This works generally but the vertical speed is about 4 times slower than the horizontal speed, which makes steering quite awkward.

I can adjust the overall mouse speed at "Core Options -> Input -> Mouse Speed" but this affects both axis. The problem seems to be more the vertical speed, because even when I crank up the mouse speed to 300% I still need almost the full screen height to move the mouse pointer from top to bottom, while horizontally it's then only about 1.5 cm for the full width. The problem seems to be with PUAE because when I use the ScummVM core mouse movement works normally.

Is there any way to fix this, like to adjust the vertical and horizontal speed seperately?

**Comment by sonninnos** [comment]
Just tested with my phone and indeed the touchscreen moves mouse differently depending on the emulated screen width and height. Default resolution (hires singleline) is exact with both dimensions, but other modes nope.

**Comment by loomes77** [comment]
Hmm, I had it all on "Automatic" actually (Automatic PAL, Resolution, Line Mode, Pixel Aspect Ratio). 

Changing the line mode to "double line" actually improves things here. Horizontal is then only double as fast as vertical. Still not perfect, but better than 4x. ;)

Changing the other settings don't seem to have an effect on the mouse speed though.

**Comment by sonninnos** [comment]
Hmm, the difference between them wasn't that drastic though. In any case it seems like this is a frontend issue, since the core only reads mouse device as mouse, so changing it would affect all mouse action. As in frontend is the one that is converting touch input to mouse input, and it does not seem play ball with the wacky aspect ratio.

How neutral is your test? Many games can't be trusted with mouse pointers, so try either Workbench or holding both mouse buttons on startup with later Kickstarts (2.0+) to get a reliable mouse cursor.

**Comment by loomes77** [comment]
Yeah, I tried Oil Imperium which loads first into Workbench, and the offset was the same in the game and Workbench. Now I also tried a pure Workbench 1.3 but it's just the same.

Another thing I noticed is that in Maniac Mansion the mouse is for same reason just about half as fast as in both Oil Imperium/Workbench, so I also have to double the mouse speed setting in PUAE here. Weird stuff. ;)

And when I load Maniac Mansion in ScummVM the horizontal speed is maybe also slightly faster than vertically, but not nearly as much as in PUAE.

As for a frontend I'm starting the games directly from Retroarch, same settings for PUAE and ScummVM.

But with the 2:1 offset in double line mode it works actually quite well, so it's not a big deal anymore. The problem was just the 4:1 offset before, which was kinda extreme. So I'm fine with that if no easy solution is possible here. :)

Anyway, I'd like to thank you for your work on PUAE!  Apart from that touchscreen issue I didn't have any problems so far and the games work really well. (I just absolutely love how those Amiga games look on the small 640x480 screen of the RP2+...)

**Comment by i30817** [comment]
This core is basically one of the best maintained cores in the entire project, so your praise is totally warranted and i will echo it.

Just the whdload loader helper and the focus on autoconfig is the kind of thing that makes the project stand above other projects.
